pkg/apis/ceph.rook.io/v1/types.go

```diff
@@ -227,9 +227,14 @@ type LogCollectorSpec struct {
 	// Enabled represents whether the log collector is enabled
 	// +optional
 	Enabled bool `json:"enabled,omitempty"`
-	// Periodicity is the periodicity of the log rotation
+	// Periodicity is the periodicity of the log rotation. For example: 1h/H, 1d/D
+	// +kubebuilder:validation:Pattern=`^(hourly|daily|weekly|monthly)$`
 	// +optional
 	Periodicity string `json:"periodicity,omitempty"`
+	// MaxLogSize is the maximum size of the log per ceph daemons. For example: 500M, 1G
+	// +kubebuilder:validation:Pattern=`^([0-9])+(?-i)(M|G)$`
```

Comment:
@travisn I  know this is something which we don't decide to do but using the above regex giving  us two advantages over `resources.Quantity`

1.  no need to translate `size` according to what is supported in logrotate like `K,M,G` and has less change to make
2. no need to handle any errors and also it will give user errors if given other formats instead of overriding the input with default.

Let's discuss offline more